nvflare/fuel/utils/config.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from collections import OrderedDict
+from enum import Enum
+from typing import Dict, Optional
+
+
+class ConfigFormat(Enum):
+    JSON = "JSON"
+    PYHOCON = "PYHOCON"
+    OMEGACONF = "OMEGACONF"
+
+    @classmethod
+    def config_ext_formats(cls):
+        return OrderedDict(
+            {
+                ".json": ConfigFormat.JSON,
+                ".conf": ConfigFormat.PYHOCON,
+                ".yml": ConfigFormat.OMEGACONF,
+                ".json.default": ConfigFormat.JSON,
+                ".conf.default": ConfigFormat.PYHOCON,
+                ".yml.default": ConfigFormat.OMEGACONF,
+            }
+        )
+
+
+class Config(ABC):
+    @abstractmethod
+    def get_native_conf(self):
+        """
+        Return the original underline config object representation if you prefer to use it directly
+        Pyhocon → ConfigTree
+        JSON → Dict
+        OMEGACONF → ConfigDict
+
+        Returns: Any, native config objects
+        """
+
+        pass
+
+    @abstractmethod
+    def get_format(self):
+        """
+        Returns: ConfigFormat, returns the current config objects ConfigFormat
+        """
+
+        pass
+
+    @abstractmethod
+    def get_location(self) -> Optional[str]:
+        """
+        Returns: None if the config is not from file else return the file path where this configuration is loaded from
+
+        """
+        pass
+
+    @abstractmethod
+    def to_dict(self) -> Dict:
+        """
+        convert underline config object to dictionary
+        Returns:Dict
+
+        """
+        pass
+
+    @abstractmethod
+    def to_conf(self, element: Dict) -> str:
+        """
+            convert dict element to the str representation of the underline configuration.
```

Comment:
Since it returns a string, suggest change method name to "to_conf_str" or "to_str" to be more clear?